backend/src/main/java/rocks/inspectit/gepard/agentmanager/connection/service/ConnectionService.java

```diff
@@ -0,0 +1,57 @@
+/* (C) 2024 */
+package rocks.inspectit.gepard.agentmanager.connection.service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import rocks.inspectit.gepard.agentmanager.connection.model.Connection;
+import rocks.inspectit.gepard.agentmanager.connection.model.dto.ConnectionDto;
+import rocks.inspectit.gepard.agentmanager.connection.model.dto.CreateConnectionRequest;
+
+/** Service-Implementation for handling agent connection requests. */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ConnectionService {
+
+  private final HashMap<UUID, Connection> connectionCache = new HashMap<>();
```

Comment:
I believe this is not thread-safe. For instance, a HashMap cannot handle a call of `handelConnectRequest()` and `getConnection()` at the same time, right?

Reply:
I guess we can then use a `ConcurrentHashMap` for now.